pkgs/development/python-modules/hydrus/default.nix

```diff
@@ -0,0 +1,59 @@
+{ stdenv, fetchFromGitHub, lzma, qt5, wrapQtAppsHook
+, buildPythonPackage, pythonPackages, isPy3k, python
+, miniupnpc_2, swftools
+}:
+
+buildPythonPackage {
+  pname = "hydrus";
+  version = "413";
+  format = "other";
+  disabled = !isPy3k;
+
+  src = fetchFromGitHub {
+    owner = "hydrusnetwork";
+    repo = "hydrus";
+    rev = "9fbed11bef499e01a6799b298bea7d0967d30430";
+    sha256 = "1dl7qpzmlxl376lzm0chmwvf4nl55wz6fwcsw0ikb33rm8r33gq4";
+  };
+
+  nativeBuildInputs = [
+    wrapQtAppsHook
+  ];
+
+  propagatedBuildInputs = with pythonPackages; [
+    beautifulsoup4 html5lib lxml numpy opencv4 pillow psutil
+    pyopenssl pyyaml requests send2trash service-identity twisted
+    lz4 lzma pysocks matplotlib qtpy pyside2 
+  ];
+
+  # tests require a graphical session
+  doCheck = false;
+
+  extraOutputsToLink = [ "doc" ];
+
+  postPatch = ''
+    sed 's;os\.path\.join(\sHC\.BIN_DIR,.*;"${miniupnpc_2}/bin/upnpc";' \
+      -i ./hydrus/core/HydrusNATPunch.py 
+
+    sed 's;os\.path\.join(\sHC\.BIN_DIR,.*;"${swftools}/bin/swfrender";' \
+      -i ./hydrus/core/HydrusFlashHandling.py
+
+  '';
+
+  installPhase = ''
+    # Move the hydrus module and related directories
+    mkdir -p $out/${python.sitePackages}
+    mv {hydrus,static} $out/${python.sitePackages}
```

Comment:
It looks like hydrus doesn't have a setup.py, so the buildPythonPackage really doesn't do much...

Reply:
I tried using the normal `mkDerivation`, but `buildPythonPackage` still does a few things that are required that I can't figure out how to cleanly replicate, such as <span><code><a href="https://nixos.org/manual/nixpkgs/stable/index.html#buildpythonpackage-function">wrapPythonPrograms</a></code></span>.

Note that I do use `format = "other";`, hopefully this is a case where that does actually make sense.

[Reference](https://nixos.org/manual/nixpkgs/stable/index.html#buildpythonpackage-parameters)